cmd/root_test.go

```diff
@@ -289,6 +291,111 @@ func TestGetAzureStackClientWithClientSecret(t *testing.T) {
 	}
 }
 
+func TestValidateAuthArgs(t *testing.T) {
```

Comment:
Thanks for adding this validation coverage!

What should we be doing if `AuthMethod` == "" ?

Reply:
It is validated by `getClient` later on, but I can a validation here as well

https://github.com/Azure/aks-engine/blob/ce337aa0ef13fccb771cf31807b6adffdf519312/cmd/root.go#L214-L219